prover/docs/05_proving_batch.md

```diff
@@ -0,0 +1,128 @@
+tools. For that, check the following [guide](./02_setup.md)(you can skip bellman-cuda step).
+
+Install the prerequisites, which you can find
+[here](https://github.com/matter-labs/zksync-era/blob/main/docs/guides/setup-dev.md). Note, that if you are not using
+Google VM instance, you also need to install [gcloud](https://cloud.google.com/sdk/docs/install#deb).
+
+Now, you can use `zk_inception` and `prover_cli` tools for setting up the env and running prover subsystem.
+
+```shell
+cargo +nightly-2024-08-01 install --git https://github.com/matter-labs/zksync-era/ --locked zk_inception zk_supervisor prover_cli --force
+```
+
+## Initializing system
+
+After you have installed the tool, you can create ecosystem(you need to run only if you are outside of `zksync-era`) by
+running:
+
+```shell
+zk_inception ecosystem create --l1-network=localhost --prover-mode=gpu --wallet-creation=localhost --l1-batch-commit-data-generator-mode=rollup --start-containers=true
+```
+
+The command will create the ecosystem and all the necessary components for the prover subsystem. You can leave default
+values for all the prompts you will see Now, you need to initialize the prover subsystem by running:
+
+```shell
+zk_inception prover init --shall-save-to-public-bucket=false --setup-database=true --use-default=true --dont-drop=false
+```
+
+For prompts you can leave default values as well.
+
+## Proving the batch
+
+### Getting data needed for proving
+
+At this step, we need to get the witness inputs data for the batch you want to prove. Database information now lives in
+input file, called `witness_inputs_<batch>.bin` generated by different core components).
+
+- If batch was produced by your system, the file is stored by prover gateway in GCS (or your choice of object storage --
+  check config). At the point of getting it, most likely there is no artifacts directory created. If you have cloned the
+  zksync-era repo, then it is in the root of ecosystem directory. Create artifacts directory by running:
+
+  ```shell
+  mkdir -p <path/to/era/prover/artifacts/witness_inputs>
+  ```
+
+  To access it from GCS (assuming you have access to the bucket), run:
+
+  ```shell
+  gsutil cp gs://your_bucket/witness_inputs/witness_inputs_<batch>.bin <path/to/era/prover/artifacts/witness_inputs>
+  ```
+
+- If you want to prove the batch produced by zkSync, you can get the data from the `ExternalProofIntegrationAPI` using
+  `{address}/proof_generation_data` endpoint. You need to replace `{address}` with the address of the API and provide
+  the batch number as a query data to get the data for specific batch, otherwise, you will receive latest data for the
+  batch, that was already proven.
+
+### Preparing database
+
+After you have the data, you need to prepare the system to run the batch. So, database needs to know about the batch and
+the protocol version it should use. You can do that with running
+
+```shell
+zk_inception prover protocol-version
```

Comment:
could you move it zk supervisor? it'd make more sense there. i'd prefer to keep inception only for initialization